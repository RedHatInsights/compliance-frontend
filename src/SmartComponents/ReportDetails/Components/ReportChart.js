import React from 'react';
import propTypes from 'prop-types';
import { Grid, GridItem } from '@patternfly/react-core';
import { ChartDonut } from '@patternfly/react-charts';
import '@/Charts.scss';
import ChartLegend from './ChartLegend';
import useDonutChart from './hooks/useDonutChart';

const ReportChart = ({ profile = {}, hasLegend = true, chartClass }) => {
  const { chartProps, legendData } = useDonutChart(profile);

  return (
    <Grid aria-label="Report chart" className={chartClass}>
      <GridItem span={hasLegend ? 6 : 12}>
        <ChartDonut {...chartProps} />
      </GridItem>
      {hasLegend ? (
        <GridItem
          span={6}
          className="pf-v5-u-display-flex pf-v5-u-align-items-center"
          style={{
            fontSize: '.85em',
            height: '100%',
          }}
        >
          <ChartLegend legendData={legendData} />
        </GridItem>
      ) : null}
    </Grid>
  );
};

ReportChart.propTypes = {
  profile: propTypes.object,
  hasLegend: propTypes.bool,
  chartClass: propTypes.string,
};

export default ReportChart;
