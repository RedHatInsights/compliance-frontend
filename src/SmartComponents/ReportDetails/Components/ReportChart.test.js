import { render } from '@testing-library/react';
import ReportChart from './ReportChart';

describe('ReportChart', () => {
  const defaultProps = {
    profile: {
      name: 'TEST Profile name',
      compliantHostCount: 0,
      testResultHostCount: 0,
      unsupportedHostCount: 0,
      totalHostCount: 0,
    },
  };

  it('expect to render without error with no profile', () => {
    const component = <ReportChart />;
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  it('expect to render without error with zero counts', () => {
    const component = <ReportChart {...defaultProps} />;
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  it('expect to render with zero unsupported hosts', () => {
    const props = {
      profile: {
        ...defaultProps.profile,
        compliantHostCount: 5,
        testResultHostCount: 5,
        unsupportedHostCount: 0,
        totalHostCount: 5,
      },
    };
    const component = <ReportChart {...props} />;
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  it('expect to render with unsupported hosts', () => {
    const props = {
      profile: {
        ...defaultProps.profile,
        compliantHostCount: 5,
        testResultHostCount: 5,
        unsupportedHostCount: 2,
        totalHostCount: 5,
      },
    };
    const component = <ReportChart {...props} />;
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  it('expect to render with not reporting hosts', () => {
    const props = {
      profile: {
        ...defaultProps.profile,
        compliantHostCount: 2,
        testResultHostCount: 5,
        unsupportedHostCount: 0,
        totalHostCount: 10,
      },
    };
    const component = <ReportChart {...props} />;
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });
});
