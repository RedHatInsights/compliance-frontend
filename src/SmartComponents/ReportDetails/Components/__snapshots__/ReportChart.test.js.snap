// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportChart expect to render with not reporting hosts 1`] = `
<div>
  <div
    class="pf-l-grid chart-container"
  >
    <div
      class="pf-l-grid__item pf-m-6-col"
    >
      <div
        class="pf-c-chart"
        style="pointer-events: none; position: relative; width: 100%; height: 100%;"
      >
        <svg
          height="230"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 230 230"
          width="230"
        >
          <g>
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
          </g>
          <text
            direction="inherit"
            dx="0"
            x="115"
            y="113.675"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 24px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); fill: #252525; stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              0
            </tspan>
            <tspan
              dx="0"
              dy="13.5"
              style="fill: #b8bbbe; font-size: 14px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              Compliant
            </tspan>
          </text>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="230"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 230 230"
            width="230"
          />
        </div>
      </div>
    </div>
    <div
      class="pf-l-grid__item pf-m-6-col pf-u-display-flex pf-u-align-items-center"
      style="font-size: .85em; height: 100%;"
    >
      <ul
        class="pf-c-list"
      >
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(0, 102, 204);"
            >
               
            </span>
          </span>
          0 systems compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(81, 157, 233);"
            >
               
            </span>
          </span>
          0 systems non-compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(244, 193, 69);"
            >
               
            </span>
          </span>
          0 systems not supported
          <span>
             
            <svg
              aria-hidden="true"
              class="grey-icon"
              fill="currentColor"
              height="1em"
              role="img"
              style="cursor: pointer;"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
              />
            </svg>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ReportChart expect to render with unsupported hosts 1`] = `
<div>
  <div
    class="pf-l-grid chart-container"
  >
    <div
      class="pf-l-grid__item pf-m-6-col"
    >
      <div
        class="pf-c-chart"
        style="pointer-events: none; position: relative; width: 100%; height: 100%;"
      >
        <svg
          height="230"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 230 230"
          width="230"
        >
          <g>
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
          </g>
          <text
            direction="inherit"
            dx="0"
            x="115"
            y="113.675"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 24px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); fill: #252525; stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              0
            </tspan>
            <tspan
              dx="0"
              dy="13.5"
              style="fill: #b8bbbe; font-size: 14px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              Compliant
            </tspan>
          </text>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="230"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 230 230"
            width="230"
          />
        </div>
      </div>
    </div>
    <div
      class="pf-l-grid__item pf-m-6-col pf-u-display-flex pf-u-align-items-center"
      style="font-size: .85em; height: 100%;"
    >
      <ul
        class="pf-c-list"
      >
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(0, 102, 204);"
            >
               
            </span>
          </span>
          0 systems compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(81, 157, 233);"
            >
               
            </span>
          </span>
          0 systems non-compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(244, 193, 69);"
            >
               
            </span>
          </span>
          0 systems not supported
          <span>
             
            <svg
              aria-hidden="true"
              class="grey-icon"
              fill="currentColor"
              height="1em"
              role="img"
              style="cursor: pointer;"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
              />
            </svg>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ReportChart expect to render with zero unsupported hosts 1`] = `
<div>
  <div
    class="pf-l-grid chart-container"
  >
    <div
      class="pf-l-grid__item pf-m-6-col"
    >
      <div
        class="pf-c-chart"
        style="pointer-events: none; position: relative; width: 100%; height: 100%;"
      >
        <svg
          height="230"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 230 230"
          width="230"
        >
          <g>
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
          </g>
          <text
            direction="inherit"
            dx="0"
            x="115"
            y="113.675"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 24px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); fill: #252525; stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              0
            </tspan>
            <tspan
              dx="0"
              dy="13.5"
              style="fill: #b8bbbe; font-size: 14px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              Compliant
            </tspan>
          </text>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="230"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 230 230"
            width="230"
          />
        </div>
      </div>
    </div>
    <div
      class="pf-l-grid__item pf-m-6-col pf-u-display-flex pf-u-align-items-center"
      style="font-size: .85em; height: 100%;"
    >
      <ul
        class="pf-c-list"
      >
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(0, 102, 204);"
            >
               
            </span>
          </span>
          0 systems compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(81, 157, 233);"
            >
               
            </span>
          </span>
          0 systems non-compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(244, 193, 69);"
            >
               
            </span>
          </span>
          0 systems not supported
          <span>
             
            <svg
              aria-hidden="true"
              class="grey-icon"
              fill="currentColor"
              height="1em"
              role="img"
              style="cursor: pointer;"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
              />
            </svg>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ReportChart expect to render without error with no profile 1`] = `
<div>
  <div
    class="pf-l-grid chart-container"
  >
    <div
      class="pf-l-grid__item pf-m-6-col"
    >
      <div
        class="pf-c-chart"
        style="pointer-events: none; position: relative; width: 100%; height: 100%;"
      >
        <svg
          height="230"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 230 230"
          width="230"
        >
          <g>
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
          </g>
          <text
            direction="inherit"
            dx="0"
            x="115"
            y="113.675"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 24px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); fill: #252525; stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              0
            </tspan>
            <tspan
              dx="0"
              dy="13.5"
              style="fill: #b8bbbe; font-size: 14px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              Compliant
            </tspan>
          </text>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="230"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 230 230"
            width="230"
          />
        </div>
      </div>
    </div>
    <div
      class="pf-l-grid__item pf-m-6-col pf-u-display-flex pf-u-align-items-center"
      style="font-size: .85em; height: 100%;"
    >
      <ul
        class="pf-c-list"
      >
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(0, 102, 204);"
            >
               
            </span>
          </span>
          0 systems compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(81, 157, 233);"
            >
               
            </span>
          </span>
          0 systems non-compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(244, 193, 69);"
            >
               
            </span>
          </span>
          0 systems not supported
          <span>
             
            <svg
              aria-hidden="true"
              class="grey-icon"
              fill="currentColor"
              height="1em"
              role="img"
              style="cursor: pointer;"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
              />
            </svg>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ReportChart expect to render without error with zero counts 1`] = `
<div>
  <div
    class="pf-l-grid chart-container"
  >
    <div
      class="pf-l-grid__item pf-m-6-col"
    >
      <div
        class="pf-c-chart"
        style="pointer-events: none; position: relative; width: 100%; height: 100%;"
      >
        <svg
          height="230"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 230 230"
          width="230"
        >
          <g>
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
            <path
              d="M5.8170722959499274e-15,-95L5.3884459162483544e-15,-88Z"
              role="presentation"
              shape-rendering="auto"
              style="fill: #d2d2d2; padding: 8px; stroke: transparent; stroke-width: 1;"
              transform="translate(115, 115)"
            />
          </g>
          <text
            direction="inherit"
            dx="0"
            x="115"
            y="113.675"
          >
            <tspan
              dx="0"
              dy="0"
              style="font-size: 24px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); fill: #252525; stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              0
            </tspan>
            <tspan
              dx="0"
              dy="13.5"
              style="fill: #b8bbbe; font-size: 14px; text-anchor: middle; font-family: var(--pf-chart-global--FontFamily); letter-spacing: var(--pf-chart-global--letter-spacing); stroke: transparent;"
              text-anchor="middle"
              x="115"
            >
              Compliant
            </tspan>
          </text>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="230"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 230 230"
            width="230"
          />
        </div>
      </div>
    </div>
    <div
      class="pf-l-grid__item pf-m-6-col pf-u-display-flex pf-u-align-items-center"
      style="font-size: .85em; height: 100%;"
    >
      <ul
        class="pf-c-list"
      >
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(0, 102, 204);"
            >
               
            </span>
          </span>
          0 systems compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(81, 157, 233);"
            >
               
            </span>
          </span>
          0 systems non-compliant
        </li>
        <li
          class="pf-c-list__item"
          style="vertical-align: middle; display: inline-block;"
        >
          <span
            class="pf-c-list__item-icon"
          >
            <span
              style="width: 10px; height: 10px; display: inline-block; background: rgb(244, 193, 69);"
            >
               
            </span>
          </span>
          0 systems not supported
          <span>
             
            <svg
              aria-hidden="true"
              class="grey-icon"
              fill="currentColor"
              height="1em"
              role="img"
              style="cursor: pointer;"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
              />
            </svg>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
